File: Exercise 3/Exercise3.cpp
// == Ejercicio 3
/* Desarrolle un algoritmo para el cálculo del salario de un trabajador. El importe
liquidado (sueldo) depende de una tarifa o precio por hora establecida y de un
condicionante sobre las horas trabajadas: si la cantidad de horas trabajadas es mayor a
40 horas, la tarifa se incrementa en un 50% para las horas extras. Calcular el sueldo
recibido por el trabajador en base las horas trabajadas y la tarifa. Utilice las
instrucciones LEER HORASTRABAJADAS y LEER TARIFA al inicio del programa para
cargar los valores en las variables HORASTRABAJADAS y TARIFA.*/

#include <iostream>
#include <sstream>

using namespace std;

// tiempo predefinido por el enunciado, en una constante global

const TIME = 40;

int main (void) {
    
    // Declaramos horas trabajadas y trarifa
    double hourPerWork, rate

    // Leemos las horas trabajadas
    cout << " Insert the hours that you work: ";
    cin >> hourPerWork;

    // leemos la tarifa
    cout << "Inser the rate per hour: "
    cin >> rate;

    // creamos una variable en la que guardaremos el sueldo final
    double salary;

    if (hourPerWork > TIME) {
        // mayor que TIME
        salary = TIME * rate + (hourPerWork - TIME ) * (1.5 * rate);
    } else {
        // menor que TIME
        salary = hourPerWork * rate;
    }

    // Mostramos el sueldo
    cout << "The salary that the worker recive is: " << salary << endl;

    return 0;
}

File: Exercise 3/Exercise3.java
// liberia para leer por entrada
import java.util.Scanner;

public class CalculateSalary {

    // Main principal
    public static void main(String[] args) {
        // Necesitamos crear el objeto scanner
        Scanner scanner = new Scanner(System.in);

        // Leemos las horas de trabajo
        System.out.print("Insert the hours that you work:");
        double hourPerWork = scanner.nextDouble();

        // Leemos la tarifa
        System.out.print("Inser the rate per hour: ");
        double rate = scanner.nextDouble();

        // creamos una variable en la que guardaremos el sueldo final
        double salary;

        if (hourPerWork > 40) {
            // mayor que 40
            salary = 40 * rate + (hourPerWork - 40) * 1.5 * rate;
        } else {
            // menos que 40
            salary = hourPerWork * rate;
        }

        System.out.println("The salary that the worker recive is" + salary);

        // Close the Scanner object
        scanner.close();
    }

}